python_modules/dagstermill/dagstermill/__init__.py

```diff
@@ -611,8 +612,8 @@ def _t_fn(transform_context, inputs):
                 )
             )
 
-            system_transform_context.events.step_materialization(
-                system_transform_context.step.key,
+            yield DagsterEvent.step_materialization(
```

Comment:
Per discussion on slack, let's make a new top-level class Materialization that is on the same "level
 as Result instead of exposing DagsterEvent